### DISCUSSION TITLE
OV5640AF

### DISCUSSION THREAD

**Discussion by littlegreenbean33** [cnadler86/micropython-camera-API#16] [Q&A]
Do I need to do any special init  with this camera ?  I bought a handful of them and none seem to be working. All is fine with OV2640 though.

**Comment by cnadler86** [comment]
Probably the AF version of this camera has a special interface that is not compatible with the driver. The OV5640 is. Take a look at the [esp32 driver](https://github.com/espressif/esp32-camera), there you will find the supported Sensors.

**Reply by cnadler86** [reply]
I took a quick look at the AF should work with the driver. Have you looked at the camera specs? The voltage might be also important, since as long as I know the ov5640 operates at 1.5v and it might need a heat disipador (see the notes in the documentation).

**Reply by littlegreenbean33** [reply]
Yes, I went trough the documentation and it seems interface is just the same. Saw that in circuitpython they mention that autofocus was working. As part of your SW is based on theirs, I assumed all would be fine.

I will do a bit more of research and post my results here. So far XIAO rejects all of the different cameras I bought. However, the seeed studio site says it is compatible.

I am going to test with an esp32cam module I have around. 

They were sold to me as esp32-cam compatible.

Let's see how far I can get.

JC

**Reply by cnadler86** [reply]
What's the error you are getting?

**Reply by cnadler86** [reply]
You could also use the esp.osdebug function to output more diagnostics. But you may need an additional uart interface for that.

**Reply by littlegreenbean33** [reply]
> What's the error you are getting?

camera not found

**Reply by littlegreenbean33** [reply]
> You could also use the esp.osdebug function to output more diagnostics. But you may need an additional uart interface for that.



If I have time along the morning I am going to try install Arduino and test one the example scripts they have for this camera.

Just to make sure there isn't anything wrong  with the cameras or that I have been sold non working knock offs

I'll keep you posted.

**Reply by littlegreenbean33** [reply]
I have been testing the cameras in Arduino and they work. They are properly recognized as OV5640 and I can take pictures with resolution beyond OV2649.

Thus, research continues....

It worries me when I saw this for AF   https://github.com/0015/ESP32-OV5640-AF/tree/main

**Reply by cnadler86** [reply]
What is the driver version of the esp32 camera driver that is used in the Arduino example?
I think this might be one difference

**Reply by littlegreenbean33** [reply]
I just did a fresh install of arduino (apt install arduino) + esp32 required libraries in order to run CameraWebServer.

Doing a find in the arduino folder I see this

./packages/esp32/tools/esp32-arduino-libs/idf-release_v5.1-632e0c2a/esp32s2/lib/libespressif__esp32-camera.a


Thus idf-release_v5.1-632e0c2a is the concrete commit.

**Reply by cnadler86** [reply]
Seems to be a pre-builded library. I took a look at the commit, but I only found out it is around 2 months old, so it is probably 2.0.13. 
If you still want, you could try building the FW with this driver version and see if it works.

**Reply by littlegreenbean33** [reply]
I am getting a bit overloaded these days. Will  have to revisit closer to the Xmas break.

**Reply by cnadler86** [reply]
Did you solve the problem? I tried it with a ov2640, 3660 and 5640. All sensors worked perfectly.

**Reply by littlegreenbean33** [reply]
Until end of the week I will not be able to revisit this problem.
